### ISSUE TITLE
Adjust progress bar for various pages of survey plugin

### ISSUE THREAD

**Issue by DominiqueMakowski** [jspsych/jsPsych#3381]
V8 brought a more granular update of the progress bar, able to change for each stimulus. That's great, and I was wondering whether it was possible to integrate it more closely with the survey plugin, and to adjust the progress bar for each "page" of a survey.

Thanks

**Comment by jodeleeuw** [comment]
@becky-gilbert maybe you have thoughts on how this might be best accomplished?

**Comment by becky-gilbert** [comment]
@DominiqueMakowski sorry that I hadn't already thought to test this! Would it work to update the progress bar manually (using [`jsPsych.setProgressBar`](https://www.jspsych.org/latest/reference/jspsych/#jspsychsetprogressbar))? If so I can provide an example of how to do this (I should add this to the docs anyway!).

Or are you and/or @jodeleeuw asking how to get jsPsych's _automatic_ progress bar to update with the `survey` pages?

**Comment by DominiqueMakowski** [comment]
The latter, I reckon a control of the JsPsych's progress bar should be automatically enabled :)

**Comment by jodeleeuw** [comment]
Given the way that the progress bar is currently implemented that would be a big overhaul. I agree that it would be nice, just want to be realistic about the chances of it happening soon 😁

**Comment by DominiqueMakowski** [comment]
No worries, but note that I think surveyJS has some callback method to get the current "page" so maybe just pulling this info would be sufficient to get the current progress within the survey, and then adjusting for the global progress

**Comment by jodeleeuw** [comment]
Yeah, I think the issue is that we would have to somehow parse the number of pages when figuring out how many units there are in the overall progress bar. Right now we do a crude count of the number of trials.
